packages/elements/src/combo-box/helpers/renderer.ts

```diff
@@ -1,30 +1,42 @@
-import type { CollectionComposer } from '@refinitiv-ui/utils/collection.js';
+import { DeprecationNotice } from '@refinitiv-ui/core';
 
-import type { Item, ItemData } from '../../item';
-import { ListRenderer } from '../../list/index.js';
+import type { CollectionComposer, DataItem } from '@refinitiv-ui/utils/collection.js';
+
+import type { ItemData } from '../../item';
+import { createListRenderer } from '../../list/index.js';
 import { Renderer } from '../../list/renderer.js';
 
+export const createComboBoxRenderer = <T extends DataItem = ItemData>(context?: unknown) => {
+  const listRenderer = createListRenderer<T>(context);
+
+  return (item: T, composer: CollectionComposer<T>, element?: HTMLElement): HTMLElement => {
+    // Extending renderer from listRenderer
+    element = listRenderer(item, composer, element);
+    const value = composer.getItemPropertyValue(item, 'value') as string;
+
+    // Using value as id for `aria-activedescendant` in combobox
+    if (value && element.id !== value) {
+      element.id = value;
+    } else if (!value && element.id) {
+      element.removeAttribute('id');
+    }
+
+    return element;
+  };
+};
+
+const deprecationNotice = new DeprecationNotice(
+  'ComboRenderer is deprecated. Use createComboBoxRenderer instead.'
+);
+
 /**
  * Renders list items as `ef-item` elements.
  * Extends its behaviour from ListRenderer.
+ * @deprecate Don't use this! It causes uncaught EvalError when using script-src 'self' CSP.
```

Comment:
Good job! We should add deprecation to jsdoc too. However, the message should be the same with deprecation notice.
```suggestion
 * @deprecate ComboRenderer is deprecated. Use createComboBoxRenderer instead.
```